features/step_definitions/routing.rb

```diff
@@ -205,3 +200,84 @@
     (page.actions_taken_table_all_actions.include? action).should == true
   end
 end
+
+
+Then /^the (.*) document should route to Groups (.*) with Group IDs (.*)$/ do |document, groups, ids|
+  # Get route log data from the edoc
+  route_log = Hash.new
+  on page_class_for(document) do |page|
+    page.show_route_log_button.wait_until_present
+    page.show_route_log unless page.route_log_shown?
+
+    page.show_pending_action_requests unless page.pending_action_requests_shown?
+    page.pnd_act_req_table_action.visible?.should
+
+    page.show_future_action_requests unless page.future_action_requests_shown?
+    page.future_actions_table.visible?.should
+
+    route_log = page.route_log_hash
+  end
+
+  # Get users represented as a single string in route log into an array
+  route_log_requested_of = route_log[:requested_of]
+  requested_users_from_route_log = Array.new
+  for i in 0..(route_log_requested_of.length - 1)
```

Comment:
This will always at least have one element, I believe (if the stuff in the parentheses evaluates to -1, you'd end up with 0..-1 for i). Maybe instead the loop should be:
```ruby
route_log_requested_of.each_with_index do |row, i|
...
```
Double-check the syntax, of course, but I think that will skip the loop if the array is empty.